smoke-test/tests/cypress/cypress/e2e/mutations/add_users.js

```diff
@@ -1,47 +1,100 @@
+        cy.visit("/signup?invite_token=bad_token");
+        tryToSignUp();
+        cy.waitTextVisible("Failed to log in! An unexpected error occurred.");
+      });
+  });
+});
 
-        cy.clickOptionWithText("Invite Users");
+describe("Reset Password", () => {
+  it("Verify you can’t generate a reset password link for a non-native user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.wait(2000);
+    cy.get("[data-testid=userItem-non-native]").first().click();
+    cy.get("[data-testid=resetButton]").should(($button) => {
+      const isDisabled = $button.prop("disabled");
+      if (Cypress.env("ADMIN_USERNAME") !== "datahub") {
+        expect(isDisabled).to.eq(true);
+      }
+    });
+  });
+
+  it("Generate a reset password link for a native user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.wait(2000);
+    cy.get("[data-testid=userItem-native]").first().click();
+    cy.get("[data-testid=resetButton]").first().click();
+    cy.get("[data-testid=refreshButton]").click();
+    cy.waitTextVisible("Generated new link to reset credentials");
+    cy.wait(2000);
 
-        cy.waitTextVisible(/signup\?invite_token=\w{32}/).then(($elem) => {
-            const inviteLink = $elem.text();
-            cy.log(inviteLink);
-            cy.visit("/settings/identities/users");
-            cy.logout();
-            cy.visit(inviteLink);
-            let name = tryToSignUp();
-            cy.waitTextVisible("Welcome to DataHub");
-            cy.hideOnboardingTour();
-            cy.waitTextVisible(name);
-        }).then(() => {
-            cy.logout();
-            cy.visit("/signup?invite_token=bad_token");
-            tryToSignUp();
-            cy.waitTextVisible("Failed to log in! An unexpected error occurred.");
-        });
+    cy.window().then((win) => {
+      cy.stub(win, "prompt");
     });
-});
+    cy.get(".ant-typography-copy").should("be.visible").click();
+    cy.get(".ant-modal-close").should("be.visible").click();
 
+    cy.waitTextVisible(/reset\?reset_token=\w{32}/)
+      .then(($elem) => {
+        const inviteLink = $elem.text();
+        cy.logout();
+        cy.visit(inviteLink);
+        cy.enterTextInTestId("email", email);
+        cy.enterTextInTestId("password", password);
+        cy.enterTextInTestId("confirmPassword", password);
+        cy.get("[type=submit]").click();
+        cy.wait(2000);
+        cy.hideOnboardingTour();
+        cy.waitTextVisible("Welcome back");
+      })
+      .then(() => {
+        cy.logout();
+        cy.visit("/reset?reset_token=bad_token");
+        cy.enterTextInTestId("email", email);
+        cy.enterTextInTestId("password", password);
+        cy.enterTextInTestId("confirmPassword", password);
+        cy.get("[type=submit]").click();
+        cy.waitTextVisible("Failed to log in!");
+      });
+  });
+});
 // Verify you can’t generate a reset password link for a non-native user (root, for example)
```

Comment:
Please remove all of these comments at the end. They were added in the beginning to just document what needed to be automated.